txtorcon/torconfig.py

```diff
@@ -445,9 +445,13 @@ def __init__(self, ports, key_blob_or_type='NEW:BEST', auth=[], ver=2):
         # FIXME nicer than assert, plz
         assert ' ' not in self._key_blob
         assert isinstance(ports, list)
-        if not key_blob_or_type.startswith('NEW:') \
-           and (len(key_blob_or_type) > 825 or len(key_blob_or_type) < 820):
-            raise RuntimeError('Wrong size key-blob')
+        try:
+            key_type, key_blob = key_blob_or_type.split(':', 1)
+        except (AttributeError, ValueError):
+            raise RuntimeError('Wrong key-blob format')
```

Comment:
> These should be ValueErrors too, IMO.

Okay. So instead of raising RuntimeErrors, raise ValueErrors?

> Personally, I: wouldn't catch AttributeError (or, if there is some sort of check basically against the type, check isinstance(key_blob_or_type, six.text_type) or whatever is appropriate).

That's certainly a better solution!

> I'd also be more explicit about the colon, something like if not ':' in ...: raise ValueError("key_blob_or_type must be a string like ...")

Hmm, I thought it would be "more Pythonic" to assume that if `split` does not return two values, then the assignment will fail with a ValueError. I will make sure to use a more explicit check then.